lib/iris/experimental/ugrid/mesh.py

```diff
@@ -2737,8 +2801,7 @@ def __init__(
         use_metadict = node_metadict.copy()
         if location != "node":
             # Location is either "edge" or "face" - get the relevant coord.
-            kwargs = {f"include_{location}s": True, "axis": axis}
-            location_coord = self.mesh.coord(**kwargs)
+            location_coord = self.mesh.coord(axis=axis, location=location)
```

Comment:
You've already made some changes but it could get a lot simpler ❤

```python
        mesh, location, axis = self.mesh, self.location, self.axis
        node_coord = mesh.coord(location="node", axis=axis)
        
        if location == "node":
            points_coord = node_coord
            bounds_connectivity = None
        else:
            points_coord = mesh.coord(location=location, axis=axis)
            bounds_connectivity = getattr(mesh, f"{location}_node_connectivity")
```